src/nvim/CMakeLists.txt

```diff
@@ -252,6 +268,17 @@ add_custom_command(OUTPUT ${GENERATED_OPTIONS}
   DEPENDS ${OPTIONS_GENERATOR} ${OPTIONS_LIST_FILE}
 )
 
+# Make NEOVIM_GENERATED_FOR_SOURCES and NEOVIM_GENERATED_FOR_HEADERS mutually exclusive
+# If a header in NEOVIM_GENERATED_FOR_HEADERS is also included in NEOVIM_GENERATED_SOURCES,
+# the header should be included only in NEOVIM_GENERATED_HEADERS
+
+foreach(hfile ${NEOVIM_GENERATED_FOR_HEADERS})
+  list(FIND NEOVIM_GENERATED_FOR_SOURCES ${hfile} hfile_idx)
+  if(NOT ${hfile_idx} EQUAL -1)
+    list(REMOVE_AT NEOVIM_GENERATED_FOR_SOURCES ${hfile_idx})
```

Comment:
My intention to rename NEOVIM_GENERATED to NEOVIM_GENERATED_FOR was to let the name consistent with the content to let developers understand them easily. It was not intended to make them mutually exclusive. If a generated header file are included in headers, it should be appended to NEOVIM_GENERATED_HEADERS, and it is also included in sources, it should also be appended to NEOVIM_GENERATED_SOURCES. That was the semantics I intended.
And, if CMake permits duplicated files to build dependency trees, then these two lists don't need to be mutually exclusive.
So, if you understand my intention and the semantics is not collapsed, I would like to remove the invariance that these two lists should be mutually exclusive.